src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_params.py

```diff
@@ -182,17 +182,26 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         # Run agent parameters
         c.argument('cpu', type=int, help='The CPU configuration in terms of number of cores required for the run.')
 
+        # MSI parameter
+        c.argument('assign_identity', nargs='*', help="Assigns managed identities to the task. Use '[system]' to refer to the system-assigned identity or a resource ID to refer to a user-assigned identity.")
+
     with self.argument_context('acr task create') as c:
         c.argument('task_name', completer=None)
 
+    with self.argument_context('acr task identity') as c:
+        c.argument('identities', options_list=['--identities'], nargs='*', help="Assigns managed identities to the task. Use '[system]' to refer to the system-assigned identity or a resource ID to refer to a user-assigned identity.")
+
     with self.argument_context('acr task credential') as c:
         # Custom registry credentials
         c.argument('login_server', help="The login server of the custom registry. For instance, 'myregistry.azurecr.io'.", required=True)
 
     for scope in ['acr task credential add', 'acr task credential update']:
         with self.argument_context(scope) as c:
-            c.argument('username', options_list=['--username', '-u'], help='The username to login to the custom registry.', required=True)
-            c.argument('password', options_list=['--password', '-p'], help='The password to login to the custom registry.', required=True)
+            c.argument('username', options_list=['--username', '-u'], help="The username to login to the custom registry.")
+            c.argument('password', options_list=['--password', '-p'], help="The password to login to the custom registry.")
+            c.argument('kv_username', help="The key vault secret URI of the username to login to the custom registry.")
+            c.argument('kv_password', help="The key vault secret URI of the password to login to the custom registry.")
```

Comment:
These should simply be folded. `--username` and `--password` should accept a keyvault URI or a value. If there is a compelling reason to keep these separate, there is (to my knowledge) no precedence for the names `--kv-username` and `--kv-password`.

Reply:
@northtyphoon @shahzzzam @sajayantony I agree with Travis here. We can simply check if either username or password contains `vault.azure.net` instead of introducing two new parameters. What do you think?

Reply:
@tjprescott see update.

Reply:
@tjprescott the team is concerned to combine `--kv-username` and `--kv-password` parameters with the username/password parameters. For instance, if the parameter has a keyvault domain in its raw value we will treat that input a keyvault URI credential and require a managed identity. Passwords can be anything. Should we constrain what the raw input can be?